cores/esp8266/HashBuilder.h

```diff
@@ -0,0 +1,44 @@
+#ifndef __ESP8266_HASH_BUILDER__
+#define __ESP8266_HASH_BUILDER__
+
+#include <WString.h>
+#include <Stream.h>
+#include <memory>
+#include <vector>
+
+class HashBuilder{
+  public:
+    void begin(void);
+    void add(uint8_t * data, uint16_t len);
+    void calculate(void);
+
+    void addHexString(const char * data);
+    bool addStream(Stream & stream, const size_t maxLen);
+    String toString(void);
+
+    // deprecated
+    void getBytes(uint8_t * output );
+    void getChars(char * output);
+
+    // INLINE
+    void add(char * data){ add((const char*)data); }
+    void add(String data){ add(data.c_str()); }
+    void addHexString(char * data){ addHexString((const char*)data); }
+    void addHexString(String data){ addHexString(data.c_str()); }
+    void add(const char * data){ add((uint8_t*)data, strlen(data)); }
+
+  protected:
+    virtual void _init(void) = 0;
+    virtual void _update(uint8_t * data, uint16_t len) = 0;
```

Comment:
I know this is carried over from the original code, but since you are improving things, would you mind replacing this uint16_t with size_t? It doesn't serve any purpose and only adds extra `extui` instructions to truncate 32-bit int to a 16-bit one.